shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -254,71 +252,118 @@ public InputConnection onCreateInputConnection(EditorInfo outAttrs) {
-        pointer.pointer = event.getPointerId(pointerIndex);
-        pointer.type = pointerType;
-        pointer.kind = PointerKind.TOUCH;
-        pointer.x = event.getX(pointerIndex);
-        pointer.y = event.getY(pointerIndex);
+        long timeStamp = event.getEventTime() * 1000; // Convert from milliseconds to microseconds.
 
-        pointer.buttons = 0;
-        pointer.down = false;
-        pointer.primary = false;
-        pointer.obscured = false;
+        packet.putLong(timeStamp); // time_stamp
+        packet.putLong(event.getPointerId(pointerIndex)); // pointer
+        packet.putLong(pointerChange); // change
+        packet.putLong(pointerKind); // kind
+        packet.putDouble(event.getX(pointerIndex)); // physical_x
+        packet.putDouble(event.getY(pointerIndex)); // physical_y
+
+        if (pointerKind == kPointerDeviceKindMouse) {
+          packet.putLong(event.getButtonState() & 0x1F); // buttons
+        } else if (pointerKind == kPointerDeviceKindStylus) {
+          packet.putLong((event.getButtonState() >> 4) & 0xF); // buttons
+        } else {
+          packet.putLong(0); // buttons
+        }
+
+        packet.putLong(0); // obscured
 
         // TODO(eseidel): Could get the calibrated range if necessary:
         // event.getDevice().getMotionRange(MotionEvent.AXIS_PRESSURE)
-        pointer.pressure = event.getPressure(pointerIndex);
-        pointer.pressureMin = 0.0f;
-        pointer.pressureMax = 1.0f;
+        packet.putDouble(event.getPressure(pointerIndex)); // presure
+        packet.putDouble(0.0); // pressure_min
+        packet.putDouble(1.0); // pressure_max
+
+        if (pointerKind == kPointerDeviceKindStylus) {
+          packet.putDouble(event.getAxisValue(MotionEvent.AXIS_DISTANCE, pointerIndex)); // distance
+          packet.putDouble(0.0); // distance_max
+        } else {
+          packet.putDouble(0.0); // distance
+          packet.putDouble(0.0); // distance_max
+        }
 
-        pointer.distance = 0.0f;
-        pointer.distanceMin = 0.0f;
-        pointer.distanceMax = 0.0f;
+        packet.putDouble(event.getToolMajor(pointerIndex)); // radius_major
+        packet.putDouble(event.getToolMinor(pointerIndex)); // radius_minor
 
-        pointer.radiusMajor = 0.0f;
-        pointer.radiusMinor = 0.0f;
-        pointer.radiusMin = 0.0f;
-        pointer.radiusMax = 0.0f;
+        packet.putDouble(0.0); // radius_min
+        packet.putDouble(0.0); // radius_max
 
-        pointer.orientation = 0.0f;
-        pointer.tilt = 0.0f;
+        packet.putDouble(event.getAxisValue(MotionEvent.AXIS_ORIENTATION, pointerIndex)); // orientation
 
-        result.add(pointer);
+        if (pointerKind == kPointerDeviceKindStylus) {
+          packet.putDouble(event.getAxisValue(MotionEvent.AXIS_TILT, pointerIndex)); // tilt
+        } else {
+          packet.putDouble(0.0); // tilt
+        }
```

Comment:
wouldn't hurt to assert() here that this routine put the correct number of fields into |packet| to isolate busted serialization changes